src/dmd/mtype.d

```diff
@@ -5498,7 +5498,7 @@ extern (C++) final class TypeStruct : Type
             if (i == 0)
             {
             }
-            else if (v.offset == offset)
+            else if (!(sym.postblit || sym.dtor) && v.offset == offset)
```

Comment:
This is rather mystifying. First off, this clause applies to unions, and the added test has nothing to do with unions. Second, the test, being on `sym`, is loop invariant and so looks like it should not be in this loop at all. Third, it is not checked if it is the first field, and the test case conveniently has the immutable member as the second field - I suspect this fix won't work at all if fields `a` and `b` are swapped. And lastly, this test looks like it should be someplace other than this function.

Reply:
It seems that a postblit should not even be generated for this struct, and that's where the fix should go.